rabbit-core/src/main/java/org/ohdsi/utilities/DateUtilities.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright 202 Observational Health Data Sciences and Informatics
+ *
+ * This file is part of WhiteRabbit
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+package org.ohdsi.utilities;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class DateUtilities {
+    /**
+     * Parses a date to millis. Copies behaviour of StringUtilities.isDate()
+     * Two recognised date formats:
+     *  - yyyy*dd*MM
+     *  - MM*dd*yy
+     * TODO: what if time included? What if datetime database field?
+     * @param dateString value to be parsed
+     * @return long date as time in millis
+     */
+    public static long parseDate(String dateString) {
+        String pattern = "";
+        if (dateString.length() == 10) {
+            char separator = dateString.charAt(4);
+            pattern = "yyyy" + separator + "MM" + separator + "dd";
+        } else if (dateString.length() == 8) {
+            char separator = dateString.charAt(2);
+            pattern = "MM" + separator + "dd" + separator + "yy";
+        }
+
+        try {
+            Date date = new SimpleDateFormat(pattern).parse(dateString);

```

Comment:
Can I suggest instead to use `java.time` package
```java
LocalDate localDate = LocalDate.from(DateTimeFormatter.ofPattern(pattern).parse(dateString))
return localDate.toEpochDay()
```
using epoch day means you don't have to do any conversions or worry about time zones. It's the number of days since the unix epoch (1970-01-01). To print out this value, simply use
```
LocalDate.fromEpochDay(epochDay).toString()
```
and it will have the `yyyy-MM-dd` format.

Reply:
I do not see how timezone is relevant, as we don't know what the timezone is of the inputted dates. But I do like the elegance of using `java.time`, so will use  it.

Reply:
The issue I’m referring to is that simpledateformatter uses the local time zone to compute the epoch time. That means that the actual values in the sample are dependent on the user’s time zone. Thinking it through though, it shouldn’t make a difference for the output, since that is also printed with a simpledateformatter with default time zone and standard deviation is independent of shift. Still good to switch to epoch day though.